From cf89664955b3641f716be3ab339eab3ea4339c50 Mon Sep 17 00:00:00 2001
From: Valentin Caron <valentin.caron@foss.st.com>
Date: Wed, 10 May 2023 12:35:34 +0200
Subject: [PATCH 0408/1695] dts: add support for HPDMA

Add HPDMA nodes definition under soc node in stm32mp251.dtsi.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 1499262abbaea5528569e7f6be04a791b30e0838 dts: add support for HPDMA]
Signed-off-by: Gatien Chevallier <gatien.chevallier@st.com>
Change-Id: Ice8d7cf3e231bf8e5a82cbdc413c6f89039ce939
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/247681
---
 core/arch/arm/dts/stm32mp251.dtsi | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index c2887f9a0..56ce67eb6 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -94,17 +94,25 @@
 		hpdma1: dma-controller@40400000 {
 			compatible = "st,stm32-dma3";
 			reg = <0x40400000 0x1000>;
-			status = "disabled";
+			clocks = <&rcc CK_BUS_HPDMA1>;
+			resets = <&rcc HPDMA1_R>;
+			#dma-cells = <4>;
 		};
 
 		hpdma2: dma-controller@40410000 {
+			compatible = "st,stm32-dma3";
 			reg = <0x40410000 0x1000>;
-			status = "disabled";
+			clocks = <&rcc CK_BUS_HPDMA2>;
+			resets = <&rcc HPDMA2_R>;
+			#dma-cells = <4>;
 		};
 
 		hpdma3: dma-controller@40420000 {
+			compatible = "st,stm32-dma3";
 			reg = <0x40420000 0x1000>;
-			status = "disabled";
+			clocks = <&rcc CK_BUS_HPDMA3>;
+			resets = <&rcc HPDMA3_R>;
+			#dma-cells = <4>;
 		};
 
 		ipcc1: mailbox@40490000 {
-- 
2.47.2

